nuhal_tiva/src/pin_tiva.c

```diff
@@ -66,6 +67,41 @@ static inline uint32_t pin_mask(uint32_t pin)
     return port_pins[number];
 }
 
+/// @brief map the pin to its corresponding AIN channel
+/// @param base - the address of the port base register for the port containing pin
+/// @param mask - a mask with all bits equal to zero except a 1 in the position
+/// of the pin number
+/// @return corresponding AIN channel, 255 if there does not exist an AIN channel
+static uint8_t get_ain_channel(uint32_t base, uint32_t mask) {
+    switch(base) {
+        case GPIO_PORTE_BASE:
+            switch(mask) {
+                case GPIO_PIN_3: return 0;
+                case GPIO_PIN_2: return 1;
+                case GPIO_PIN_1: return 2;
+                case GPIO_PIN_0: return 3;
+                case GPIO_PIN_5: return 8;
+                case GPIO_PIN_4: return 9;
+                default: return 255;
```

Comment:
Should this be an error() rather than returning 255 and then checking for an error later?